src/AppInstallerRepositoryCore/Microsoft/Schema/IPortableIndex.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "SQLiteWrapper.h"
+#include "Microsoft/Schema/Version.h"
+#include "Portable_1_0/PortableTable.h"
+#include <AppInstallerVersions.h>
+
+namespace AppInstaller::Repository::Microsoft::Schema
+{
+    // Forward declarations
+    struct Version;
+
+    struct IPortableIndex
+    {
+        virtual ~IPortableIndex() = default;
+
+        // Version 1.0
+        // 
+        // Gets the schema version that this index interface is built for.
+        virtual Schema::Version GetVersion() const = 0;
+
+        // Creates all of the version dependent tables within the database.
+        virtual void CreateTable(SQLite::Connection& connection) = 0;
+
+        // Adds a portable file to the index.
+        virtual SQLite::rowid_t AddPortableFile(SQLite::Connection& connection, const Portable_V1_0::PortableFile& file) = 0;

```

Comment:
The metadata about a portable file should be defined in the public area and always be the latest version.  The code in the version specific will extract only the information that it needs.

Reply:
To expand on this, the versioned schema headers should not be public.

Reply:
changed versioned schema headers to be private and moved the portable file struct to IPortableIndex